### PR TITLE
📝: Add push notification demo design

### PR DESCRIPTION
**Pull request by creature-water-valley**
## ✅ What's done

- [x] Add push notification demo design

---

## Other (messages to reviewers, concerns, etc.)

none

### CODE REVIEW

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+![permission-dialog](./ios-permission-dialog.png)
+
+### Push通知受信の仮許可
+
+iOS12以降の場合は、上述した許可ダイアログでユーザが許可をしなくても、Push通知を受信できる機能があります。
+Push通知受信のリクエスト時（許可ダイアログ表示リクエスト時）に[provisional](https://rnfirebase.io/messaging/ios-permissions#permission-settings)を設定すると、
+許可ダイアログは表示されずステータスが`PROVISIONAL`になります。
+
+ステータスが`PROVISIONAL`の場合はPush通知を受信できます。Push通知を受信した際にヘッドアップ通知は表示されませんが、通知センターには表示されるようになります。
```
Comment (in-a-paddy) [inline website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:47]: 「ヘッドアップ通知」って用語が伝わるかわからんので、関連用語を簡単に上の方で整理しておいたほうが良さそう。

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+| アプリの状態 | iOS | Android |
+|:--|:--|:--|
+|アプリがフォアグラウンドにある場合のPush通知受信|![ios-foreground](./ios-foreground.png)|![android-foreground](./android-foreground.png)|
+|アプリがバックグラウンドにある場合、もしくはアプリが停止中の場合のPush通知受信<br/>（Androidは`emergencyChannel`に送信）|![ios-background](./ios-background.png)|![andorid-background-emergency-channel](./android-background-emergency-channel.png)|
+|アプリがバックグラウンドにある場合、もしくはアプリが停止中の場合のPush通知受信<br/>（Androidの`highChannel`に送信）|-|![android-background-high-channel](./android-background-high-channel.png)|
+
+### Push通知の設定画面への遷移
+
+OSのアプリの設定画面に遷移します。
```
Comment (in-a-paddy) [inline website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:81]:
```suggestion
OSのアプリ設定画面に遷移します。
```

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx (hunk 1 above)
Comment (creature-water-valley) [reply website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:54]: Add glossary
277637a9463a9b42deb4510e4ae30a1e1a95858f

**Review by kozake — APPROVED** [review]

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# プッシュ通知
+
+## 概要
+
+プッシュ通知の機能を検証するデモページです。
```
Comment (in-a-paddy) [inline website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:5]: I feel we need a diagram to organize how users receive push notifications....
It is very complicated and difficult...
(can be anothor task)

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+## プッシュ通知の送信
+
+バックエンドのAPIを実行してプッシュ通知を送信します。
+プッシュ通知を送信するバックエンドのソースコードは[こちら](https://github.com/{@inject:organization}/mobile-app-crib-notes/tree/master/example-app/santoku-app-backend/src/main/java/jp/fintan/mobile/santokuapp/infrastructure/service/FcmPushNotifier.java)です。
+
+アプリがフォアグラウンドにある場合は、アプリが用意しているスナックバーに通知内容を表示します。
+アプリがバックグラウンドにある場合、もしくはアプリを起動していない場合は、OSによって挙動が変わります。
+
+iOSの場合は、ヘッドアップ通知が表示され、音が鳴ります。
```
Comment (in-a-paddy) [inline website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:65]: Please created tasks to introduce the changes in iOS 15. (focus mode, interruption level)

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+## 概要
+
+プッシュ通知の機能を検証するデモページです。
+
+- プッシュ通知の許可状態取得
+- プッシュ通知受信の許可ダイアログ表示
+- プッシュ通知受信の仮許可
+- プッシュ通知の送信
+- プッシュ通知の設定画面への遷移
```
Comment (in-a-paddy) [inline website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:11]: Please create tasks to add section introducing the notification center ui, including collupsed(Android), summarized(iOS), time-sensitive(iOS).

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx (hunk 3 above)
Comment (creature-water-valley) [reply website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:5]: Added!
https://dev.azure.com/ws-4020/develop-mobile-app-casually/_backlogs/backlog/Mobile%20Dev%20Team/Epics/?workitem=16375

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx (hunk 4 above)
Comment (creature-water-valley) [reply website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:65]: Added!
https://dev.azure.com/ws-4020/develop-mobile-app-casually/_backlogs/backlog/Mobile%20Dev%20Team/Epics/?workitem=16375

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx (hunk 5 above)
Comment (creature-water-valley) [reply website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:11]: Added!
https://dev.azure.com/ws-4020/develop-mobile-app-casually/_backlogs/backlog/Mobile%20Dev%20Team/Epics/?workitem=16375

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx (hunk 3 above)
Comment (in-a-paddy) [reply website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:5]: [test]
AB#16375

File: website/docs/react-native/santoku/design/demo/push-notification/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+iOSの場合は、以下4つのステータスが存在します。
+
+| ステータス | 説明 |
+|:--|:--|
+| NOT_DETERMINED | ユーザが許可ダイアログでプッシュ通知受信を許可、もしくは拒否をしていない状態 |
+| DENIED | ユーザが許可ダイアログでプッシュ通知受信を拒否した場合 |
+| AUTHORIZED | ユーザが許可ダイアログでプッシュ通知受信を許可した場合 |
+| PROVISIONAL | アプリがPROVISIONALとしてプッシュ通知受信の許可リクエストを実施した場合 |
+
+Androidの場合は、常に`AUTHORIZED`となります。
```
[the remark this answers is not in the document]
Comment (creature-water-valley) [reply website/docs/react-native/santoku/design/demo/push-notification/overview.mdx:35]: Added!
https://dev.azure.com/ws-4020/develop-mobile-app-casually/_backlogs/backlog/Mobile%20Dev%20Team/Epics/?workitem=16375

**Review by in-a-paddy — APPROVED** [review]

**Pull request merged**